File: core/src/main/java/heast/core/network/packets/c2s/auth/ResetC2SPacket.java
package heast.core.network.packets.c2s.auth;

import heast.core.network.Packet;
import heast.core.network.PacketBuf;
import heast.core.network.listeners.ServerAuthListener;

public final class ResetC2SPacket implements Packet<ServerAuthListener> {

    private final String email;
    private final String password;

    public ResetC2SPacket(String email, String password) {
        this.email = email;
        this.password = password;
    }

    public ResetC2SPacket(PacketBuf buf) {
        this.email = buf.readString();
        this.password = buf.readString();
    }

    @Override
    public void write(PacketBuf buf) {
        buf.writeString(email);
        buf.writeString(password);
    }

    @Override
    public void apply(ServerAuthListener listener) {
        listener.onReset(this);
    }

    public String getEmail() {
        return email;
    }

    public String getPassword() {
        return password;
    }
}

File: Core/Network/Packets/C2S/ResetC2SPacket.cs
﻿using Core.Network.Listeners;

namespace Core.Network.Packets.C2S;

public partial class ResetC2SPacket : AbstractPacket
{
    public ResetC2SPacket(string email, string password)
    {
        Email = email;
        Password = password;
    }

    public string Email { get; }
    public string Password { get; }

    public override void Apply(IPacketListener listener)
    {
        if (listener is IServerAuthListener authListener)
        {
            authListener.OnReset(this);
        }
    }
}